core/src/main/java/com/linecorp/armeria/common/RequestContextAccessor.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common;
+
+import org.reactivestreams.Subscription;
+
+import com.linecorp.armeria.internal.common.RequestContextUtil;
+
+import io.micrometer.context.ContextRegistry;
+import io.micrometer.context.ContextSnapshot;
+import io.micrometer.context.ContextSnapshot.Scope;
+import io.micrometer.context.ThreadLocalAccessor;
+
+/**
+ * This class works with the
+ * <a href="https://docs.micrometer.io/context-propagation/reference/index.html">
+ * Context-propagation</a> library and keep the {@link RequestContext} during
+ * <a href="https://github.com/reactor/reactor-core">Reactor</a> operations.
+ * Get the {@link RequestContextAccessor} to register it to the {@link ContextRegistry}.
+ * Then, {@link ContextRegistry} will use {@link RequestContextAccessor} to
+ * propagate context during the
+ * <a href="https://github.com/reactor/reactor-core">Reactor</a> operations
+ * so that you can get the context using {@link RequestContext#current()}.
+ * However, please note that you should include Mono#contextWrite(ContextView) or
+ * Flux#contextWrite(ContextView) to end of the Reactor codes.
+ * If not, {@link RequestContext} will not be keep during Reactor Operation.
+ */
+public final class RequestContextAccessor implements ThreadLocalAccessor<RequestContext> {
```

Comment:
Question: Is there any reason that you put this class in the core module instead of the reactor module?

Reply:
@minwoox nim, thanks for your comment 🙇‍♂️ 
Ok, i'll move this class to `com.linecorp.armeria.internal.common`;

> Question: Is there any reason that you put this class in the core module instead of the reactor module?

In the previous comment, `trustin` nim did recommend to move it! 
IMHO, `trustin` nim want to use `micro-meter:context-propagation` as common library with `Java SPI interface`. 
For example, if we implement `ContextAccessor` interface and `Context` corresponding to it, we can apply `micro-meter:context-propagation` to any `Context` to propagate.

Reply:
> Question: Is there any reason that you put this class in the core module instead of the reactor module?

In the previous comment, `trustin` nim did recommend to move it! 
IMHO, `trustin` nim want to use `micro-meter:context-propagation` as common library with `Java SPI interface`. 
For example, if we implement `ContextAccessor` interface and `Context` corresponding to it, we can apply `micro-meter:context-propagation` to any `Context` to propagate.

Reply:
Thanks for your comments and No problem at all 🙇‍♂️ 
I keep it as is! thanks a lot even if holiday 👍